### DISCUSSION TITLE
Regarding Empty String

### DISCUSSION THREAD

**Discussion by shubhamaydas** [fnando/i18n#11] [Q&A]
is there a way to removeEmptyString in the translation so that it doesn't consider that and shows default message or goes to other scope ?

**Comment by fnando** [comment]
Can you please give an example with code of what you mean?

**Reply by shubhamaydas** [reply]
we have a key in the translation file but with empty value , I want even in this scenario it should fall to en.ts as de.ts has the key but value of it is empty

title={I18n.t(‘pin’, { removeEmptyString: true })}

the en.ts
export default { pin: ‘PIN’ }

de.ts
export default {pin: ''}

**Comment by fnando** [comment]
Unfortunately, that's not something I think it should be added straight to the lib. What you can do, though, is removing the empty translation from the `i18n.translations` object.

```
> i18n.store({en: {hello: ""}});
undefined

> i18n.translations
{ en: { hello: '' } }

> i18n.t("hello")
''

> delete i18n.translations.en.hello;
true

> i18n.t("hello")
'[missing "en.hello" translation]'
```

Rather than doing it manually, you can create a function that filter out empty strings before storing it with `i18n.store(translations)`. The easiest way, imo, would using Lodash's `unset` function, case receive a path like `a.b.c`, so you can iterate on your supported locales and unset those paths.
